SU2_CFD/src/variables/CIncEulerVariable.cpp

```diff
@@ -90,12 +91,13 @@ CIncEulerVariable::CIncEulerVariable(su2double pressure, const su2double *veloci
 
   Primitive.resize(nPoint,nPrimVar) = su2double(0.0);
 
-  /*--- Incompressible flow, gradients primitive variables nDim+4, (P, vx, vy, vz, T, rho, beta),
-        We need P, and rho for running the adjoint problem ---*/
+  /*--- Incompressible flow, gradients primitive variables nDim+4, (P, vx, vy, vz, T, rho, beta) ---*/
 
-  Gradient_Primitive.resize(nPoint,nPrimVarGrad,nDim,0.0);
+  if (config->GetMUSCL_Flow() || viscous) {
+    Gradient_Primitive.resize(nPoint,nPrimVarGrad,nDim,0.0);
+  }
 
-  if (config->GetReconstructionGradientRequired()) {
+  if (config->GetMUSCL_Flow() && config->GetReconstructionGradientRequired()) {
```

Comment:
In which case is one true and the other isn't? MUSCL and ReconstructedGradient are close friends and I am unaware when they show up separated?

Reply:
You are right, I had another look at the long config post processing... what can happen is an unfortunate mix of centered and upwind options.

Reply:
And also on coarse multigrid levels, there is no MUSCL but the reconstruction will still be true, but that is too many if's to deal with in CVariable.